ARTstor.js

```diff
@@ -1,131 +1,899 @@
+                // Don't capture data if image group window is in editing state.
+                skipThumbnail = true;
+            }
+            if ((doc.getElementById("ssContentWrap") != null) && (doc.getElementById("ssContentWrap").style.display == "inline")) {
+                // Don't capture data if slide show window is present
+                skipThumbnail = true;
+            }
+            // Allow thumbnail window.
+            if (!skipThumbnail) {
+                doThumbnails(doc, url);
+            }
+        }
+    }
 }
 
-function associateData(newItem, value, field) {
-	if (!value) return;
-	if (typeof value == 'string') value = ZU.trimInternal(value);
-	if (!value) return;
-	
-	newItem[field] = value;
+function doImageViewer(doc, url) {
+    // get the image id and object type from the page
+    // this contains the objId and object type separate by : as in "AWSS35953_35953_25701160:11"
+    var objID = doc.getElementById("objID");
+    if (objID != null) {
+        var objItems = [];
+        var objItem = doc.getElementById("objID").title;
+        objItems.push(objItem);
+        processObjects(doc, url, objItems);
+    }
 }
 
-function doWeb(doc, url) {
-	var newItem = new Zotero.Item("artwork"),
-		data = getData(doc),
-		content;
-	
-	if (data.Creator) {
-		content = data.Creator.replace(/\s*\(.*/, '');
-		associateData(newItem,
-			ZU.cleanAuthor(content, "artist", artist.indexOf(',') != -1),
-			'creators'
-		);
-	}
-	
-	if (data.Date) {
-		content = data.Date.replace(/^Photographed (?:in )?/, '');
-		associateData(newItem, content, "date");
-	}
-	
-	if (data.Rights) {
-		var m = data.Rights.match(/^.*(©.*)$/m);
-		if (m) associateData(newItem, m[1], "rights");
-	}
-	
-	if (data.Subject) {
-		content = data.Subject.split(/\s*;\s*/);
-		associateData(newItem, content, 'tags');
-	}
-	
-   //these might not be complete - it's pretty straightforward to add more
-	associateData(newItem, data.Title, "title");
-	associateData(newItem, data.Measurements, "artworkSize");
-	associateData(newItem, data.Material, "artworkMedium");
-	associateData(newItem, data.Repository, "archive");
-	associateData(newItem, data['ID Number'], "archiveLocation");
-	associateData(newItem, data.Description, "abstractNote");
-	
-	newItem.complete();
-}
\ No newline at end of file
+function doMetadataWindow(doc, url) {
+    // get object id from metadata window.
+    var metaWindows = doc.getElementsByClassName('MetaDataWidgetRoot');
+    var objItems = [];
+    for (var i = 0; i < metaWindows.length; i++) {
+        var id = metaWindows[i].id.substring(3);
+        objItems.push(id);
+    }
+
+    processSelectedObject(doc, url, objItems, 1);
+}
+
+function processSelectedObject(doc, url, selectedObjs, selectionType) {
+    var serviceURL = getThumbnailServiceURL(doc, url);
+    Zotero.Utilities.HTTP.doGet(serviceURL, function(text) {
```

Comment:
The DOM in the webpage does not have all the information need for prompting the list to the user (it needs the object id, and object type that is associated with the title, which is not in the DOM). This is why it need to use the service call to get the data in javascript to show it. The issues with AJAX application is that all data are stored in javascript, not in the DOM. This is why we have to have the service call to get the correct data.